<launch>
    <!-- Load controller configuration to parameter server -->
    <rosparam file="$(find cute_bringup)/config/cute_controllers.yaml" command="load"/>

    <!-- start specified joint controllers -->
    <node name="dynamixel_controller_spawner" pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port=pan_tilt_port
                joint1_controller
                joint2_controller
                joint3_controller
                joint4_controller
                joint5_controller
                joint6_controller
                joint7_controller
                claw_controller"
          output="screen"/>

    <rosparam file="$(find cute_bringup)/config/cute_servo_manager.yaml" command="load"/>
    <node name="dynamixel_joint_states_manager" pkg="cute_bringup" type="dynamixel_joint_state_manager.py" output="screen"/>
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find cute_model)/robots/cute_300es.urdf.xacro'" />
  
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
    
    <include file="$(find cute_ros_control)/launch/cute_ros_control.launch"/>
    
    <node name="robot_arm_dynamixel_manager" pkg="cute_bringup" type="robot_arm_dynamixel_manager.py"/>
</launch>
